locksmith/src/controllers/v2/authController.ts

```diff
@@ -120,6 +121,48 @@ export const login: RequestHandler = async (request, response) => {
   }
 }
 
+export const loginWithPrivy: RequestHandler = async (request, response) => {
```

Comment:
the idea here is to first verify the access token via privy, and then proceed to obtian the idtoken (via cookies) to retrieve the user's basic details, which includes their wallet address (if available)

messy and still a draft.